crawl-ref/source/art-func.h

```diff
@@ -1274,25 +1274,55 @@ static void _OCTOPUS_KING_world_reacts(item_def *item)
 
 ///////////////////////////////////////////////////
 
+static void _CAPTAIN_equip(item_def *item, bool *show_msgs, bool unmeld)
+{
+    if (you_worship(GOD_SHINING_ONE))
+        {
+        _equip_mpr(show_msgs,
+                   "You feel dishonorable wielding this.");
+        }
+    else
+        {
+        _equip_mpr(show_msgs,
+                   "You feel a cutthroat vibe.");
+        }
+}
+
 static void _CAPTAIN_melee_effects(item_def* weapon, actor* attacker,
-                                   actor* defender, bool mondied, int dam)
+                                actor* defender, bool mondied, int dam)
 {
+    // Use a formula that isn't based on damage for tier 1 and 2 stabs
+    const bool stab = (defender->paralysed() || defender->asleep()
+                       || defender->caught() || defender->petrified()
+                       || defender->petrifying());
+
     // Player disarming sounds like a bad idea; monster-on-monster might
     // work but would be complicated.
-    if (!attacker->is_player() || !defender->is_monster() || mondied)
-        return;
-
-    if (x_chance_in_y(dam, 75))
+    if ((coinflip()
+        && dam >= (4 + random2(defender->get_hit_dice()))
+        && attacker->is_player()
+        && defender->is_monster()
+        && !mondied)
+        && (!stab || stab && one_chance_in(3)))
     {
         item_def *wpn = defender->as_monster()->disarm();
         if (wpn)
         {
-            mprf("You knock %s %s to the ground with your cutlass!",
-                 apostrophise(defender->name(DESC_THE)).c_str(),
-                 wpn->name(DESC_PLAIN).c_str());
+            mprf("You disarm %s, knocking %s onto the floor!",
+                defender->name(DESC_THE).c_str(),
+                wpn->name(DESC_A).c_str());
+            did_god_conduct(DID_UNCHIVALRIC_ATTACK, 3);
+            if ((!x_chance_in_y(defender->get_hit_dice(), random2(20) + dam*4)
+            && !stab) || (stab && !one_chance_in(3)))
```

Comment:
`&&` should be aligned with the previous line, and the `||` etc. should be on its own line